stacker/providers/aws/default.py

```diff
@@ -94,6 +94,9 @@ def __init__(self, region, **kwargs):
         # Necessary to deal w/ multiprocessing issues w/ sharing ssl conns
         # see: https://github.com/remind101/stacker/issues/196
         self._pid = os.getpid()
+        # If region is not set explicitly, fetch it from the boto session
+        if not self.region:
```

Comment:
I think you could combine this with the region line by using:

```
self.region = region or get_session().region_name
```

That said, after digging into this, I'm not really sure how get_session without any arguments is going to return an actual region_name?